ldgallery: build fails on Darwin

The build of the [ldgallery-viewer package] seems to fail on Darwin.
This seems to be due to a missing native dependency only for that platform.

[ldgallery-viewer package]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/graphics/ldgallery/viewer/default.nix

[Build log on Hydra](https://nix-cache.s3.amazonaws.com/log/76jdsjmfnnddvi5l1qwmfkr2547d5pd1-node_ldgallery-viewer-2.0.0.drv):

```
trying to install from sub 'node_module' directory, skipping Git hooks installation
...............] \ : info lifecycle yorkie@2.0.0~install: yorkie@2.0.0[0m
> ejs@2.7.4 postinstall /nix/store/rfhp2asjaf6m6qhk713dkl59wihm8mpz-node_ldgallery-viewer-2.0.0/lib/node_modules/ldgallery-viewer/node_modules/ejs
> node ./postinstall.js

Thank you for installing EJS: built with the Jake JavaScript build tool (https://jakejs.com/)

...............] \ : info lifecycle ejs@2.7.4~postinstall: ejs@2.7.4[0m
> fsevents@1.2.12 install /nix/store/rfhp2asjaf6m6qhk713dkl59wihm8mpz-node_ldgallery-viewer-2.0.0/lib/node_modules/ldgallery-viewer/node_modules/fsevents
> node-gyp rebuild

No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

make: Entering directory '/nix/store/rfhp2asjaf6m6qhk713dkl59wihm8mpz-node_ldgallery-viewer-2.0.0/lib/node_modules/ldgallery-viewer/node_modules/fsevents/build'
  SOLINK_MODULE(target) Release/.node
  CXX(target) Release/obj.target/fse/fsevents.o
../fsevents.cc:10:10: fatal error: 'CoreServices/CoreServices.h' file not found
#include "CoreServices/CoreServices.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [fse.target.mk:127: Release/obj.target/fse/fsevents.o] Error 1
make: Leaving directory '/nix/store/rfhp2asjaf6m6qhk713dkl59wihm8mpz-node_ldgallery-viewer-2.0.0/lib/node_modules/ldgallery-viewer/node_modules/fsevents/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/nix/store/a23q52llhd17vyp7n9rd256jm9xljm5g-nodejs-12.22.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Darwin 17.7.0
gyp ERR! command "/nix/store/a23q52llhd17vyp7n9rd256jm9xljm5g-nodejs-12.22.1/bin/node" "/nix/store/a23q52llhd17vyp7n9rd256jm9xljm5g-nodejs-12.22.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /nix/store/rfhp2asjaf6m6qhk713dkl59wihm8mpz-node_ldgallery-viewer-2.0.0/lib/node_modules/ldgallery-viewer/node_modules/fsevents
gyp ERR! node -v v12.22.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fsevents@1.2.12 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the fsevents@1.2.12 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /private/tmp/nix-build-node_ldgallery-viewer-2.0.0.drv-0/.npm/_logs/2021-05-09T16_23_02_686Z-debug.log

builder for '/nix/store/76jdsjmfnnddvi5l1qwmfkr2547d5pd1-node_ldgallery-viewer-2.0.0.drv' failed with exit code 1
```

I believe that adding `CoreServices` to the build inputs of the native Node dependencies could solve the issue.
However, I don't have access to any Darwin machine to test that.

Something like:
```patch
diff --git a/pkgs/tools/graphics/ldgallery/viewer/default.nix b/pkgs/tools/graphics/ldgallery/viewer/default.nix
index 9559120069f..3730c9a2dec 100644
--- a/pkgs/tools/graphics/ldgallery/viewer/default.nix
+++ b/pkgs/tools/graphics/ldgallery/viewer/default.nix
@@ -1,4 +1,8 @@
-{ lib, stdenv, fetchFromGitHub, pkgs, nodejs-12_x, pandoc }:
+{ lib, stdenv, fetchFromGitHub, pkgs, nodejs-12_x, pandoc,
+
+  # Darwin-specific
+  CoreServices
+}:

 with lib;

@@ -24,6 +28,7 @@ let
   nodePkg = nodePackages.package.override {
     src = "${sourcePkg}/viewer";
     postInstall = "npm run build";
+    buildInputs = lib.optionals stdenv.isDarwin [ CoreServices ];
   };

 in
```

--- a/pkgs/tools/graphics/ldgallery/default.nix
+++ b/pkgs/tools/graphics/ldgallery/default.nix
@@ -1,10 +1,10 @@
-{ lib, pkgs, makeWrapper, haskellPackages, haskell, pandoc, imagemagick }:
+{ lib, pkgs, makeWrapper, haskellPackages, haskell, pandoc, imagemagick, CoreServices }:
 
 with lib;
 with haskell.lib;
 
 let
-  ldgallery-viewer = pkgs.callPackage ./viewer { };
+  ldgallery-viewer = pkgs.callPackage ./viewer { inherit CoreServices; };
   inherit (haskellPackages) ldgallery-compiler;
 
 in

--- a/pkgs/tools/graphics/ldgallery/viewer/default.nix
+++ b/pkgs/tools/graphics/ldgallery/viewer/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, pkgs, nodejs-12_x, pandoc }:
+{ lib, stdenv, fetchFromGitHub, pkgs, nodejs-12_x, pandoc, CoreServices }:
 
 with lib;
 
@@ -24,6 +24,7 @@ let
   nodePkg = nodePackages.package.override {
     src = "${sourcePkg}/viewer";
     postInstall = "npm run build";
+    buildInputs = optionals stdenv.isDarwin [ CoreServices ];
   };
 
 in

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6032,7 +6032,9 @@ in
 
   ldc = callPackage ../development/compilers/ldc { };
 
-  ldgallery = callPackage ../tools/graphics/ldgallery { };
+  ldgallery = callPackage ../tools/graphics/ldgallery {
+    inherit (darwin.apple_sdk.frameworks) CoreServices;
+  };
 
   lbreakout2 = callPackage ../games/lbreakout2 { };
 

